server/controllers/threads.controller.js

```diff
@@ -0,0 +1,61 @@
+import httpStatus from 'http-status';
+import Promise from 'bluebird';
+import db from '../../config/sequelize';
+import APIError from '../helpers/APIError';
+
+const Message = db.Message;
+const Thread = db.Thread;
+const User = db.User;
+
+/**
+ * Start a new thread
+ * @property {Array} req.body.participants - Array of usernames to include in the thread.
+ * @property {string} req.body.subject - Subject line of the message
+ * @property {string} req.body.message - Body of the message
+ * @property {string} req.body.topic - Topic of the thread
+ * @returns {Message}
+ */
+function create(req, res, next) {
+    const participants = req.body.participants;
+    console.log("found participant", participants)
+    let users = [];
+    let userPromises = [];
+    participants.forEach((participant) => {
+      let userPromise = User.findOrCreate({where: {username: participant}})
+      .spread((user, created) => {
```

Comment:
@ElijahGeorge so you are using .spread here because sequelize's `findOrCreate` returns an array?

Reply:
Yes @puckybreg. `created` is a `boolean` that indicates whether the returned user was created or just found in the database. It's actually different from the spread operator `...`. I'm not sure why they decided to name it that 😕